tools/storybook/amp-env/decorator.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {h} from 'preact'; // eslint-disable-line
+import render from 'preact-render-to-string/jsx';
+
+import {makeDecorator} from '@storybook/addons';
+
+export default makeDecorator({
+  name: 'withAmpDecorator',
+  parameterName: 'amp',
+  wrapper: (getStory, context) => {
+    const contents = render(getStory(context));
+
+    const ampHtml = `
+        <!doctype html>
+        <html amp lang="en">
+        <head>
+            <meta charSet="utf-8" />
+            <title>AMP Page Example</title>
+            <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1" />
+            <script async src="https://cdn.ampproject.org/v0.js"></script>
+            <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+            ${(context?.parameters?.extensions || []).map(
+              ext =>
+                `<script async custom-element="${
```

Comment:
Some of the extensions are templates. But we can worry about this when we get there.